DependencyInjection/SonataUserExtension.php

```diff
@@ -307,5 +316,23 @@ private function checkManagerTypeToModelTypeMapping(array $config)
                 );
             }
         }
+
+        foreach ($actualAdminClasses as $actualAdminClass) {
+            if ('\\' === substr($actualAdminClass, 0, 1)) {
+                $actualAdminClass = substr($actualAdminClass, 1);
+            }
+
+            foreach ($disallowedAdminClasses as $disallowedAdminClass) {
+                if ($actualAdminClass === $disallowedAdminClass || is_subclass_of($actualAdminClass, $disallowedAdminClass)) {
+                    throw new \InvalidArgumentException(
+                        sprintf(
+                            'Admin class "%s" does not correspond to manager type "%s".',
+                            $actualAdminClass,
+                            $managerType
+                        )
+                    );
+                }
+            }
+        }
```

Comment:
I don't think that check is very useful, and it doesn't scale well.

Reply:
Actually yes. It is better to remove it?

Reply:
Yeah. Or maybe introduce marker interfaces and trigger a deprecation if they are not implemented, but meh

Reply:
Ok. I'm removing :)